.github/workflows/reusable-test.yml

```diff
@@ -259,7 +279,7 @@ jobs:
         files: c:/dumps/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}/*.dmp
 
     - name: Upload any crash dumps
-      if: (failure()) && (inputs.gather_dumps == true)
+      if: always() && (steps.check_dumps.outputs.files_exists == 'true') && (inputs.gather_dumps == true)
```

Comment:
doesnt always() evaluate to TRUE always?

Reply:
GitHub's workflow language is weird:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-not-requiring-successful-dependent-jobs

Nothing or true -> Only run on success.
Failure -> Only run on failure.
Always -> Run in either success or failure.